core/pta/bcm/elog.c

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2019 Broadcom.
+ */
+
+#include <drivers/bcm/bnxt.h>
+#include <io.h>
+#include <kernel/pseudo_ta.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <string.h>
+#include <trace.h>
+
+#define ELOG_SERVICE_UUID \
+		{ 0x6272636D, 0x2019, 0x0701, \
+		{ 0x42, 0x43, 0x4D, 0x5F, 0x45, 0x4C, 0x4F, 0x47 } }
+
+#define ELOG_TA_NAME			"pta_bcm_elog.ta"
+
+#define BCM_NITRO_FW_LOAD_ADDR			0x8ae00000
+#define BCM_NITRO_CRASH_DUMP_BASE_ADDR		0x8b000000
+
+/* Default ELOG buffer size 1MB */
+#define DEFAULT_ELOG_BUFFER_SIZE		0x100000
+
+/*
+ * Get Error log memory dump
+ *
+ * [out]    memref[0]:    Destination
+ * [in]     value[1].a:   Offset
+ */
+#define PTA_BCM_ELOG_CMD_GET_ELOG_MEM		1
+
+/*
+ * Get nitro crash_dump memory
+ *
+ * [out]    memref[0]:    Destination
+ * [in]     value[1].a:   Offset
+ */
+#define PTR_BCM_ELOG_CMD_GET_NITRO_CRASH_DUMP	2
+
+/*
+ * Load nitro firmware memory
```

Comment:
This PTA is already handling secure nitro memory block, firmware is also part of this block.
also I wanted avoid having another PTA and user space application just for loading firmware

Reply:
For now will keep the name as is. Since I need to decouple the changes for getting and loading memory from application end also. In future commits, I'll try to decouple and name PTAs accordingly.

Reply:
OK